packages/fast-usdc/src/clientSupport.js

```diff
@@ -0,0 +1,119 @@
+import { assertAllDefined } from '@agoric/internal';
+
+/**
+ * @import {USDCProposalShapes} from './pool-share-math.js';
+ */
+
+/**
+ * @param {Pick<
+ *   import('@agoric/vats/tools/board-utils.js').AgoricNamesRemotes,
+ *   'brand'
+ * >} agoricNames
+ * @param {object} opts
+ * @param {string} opts.offerId
+ * @param {bigint} opts.fastLPAmount
+ * @param {bigint} opts.usdcAmount
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+const makeDepositOffer = ({ brand }, { offerId, fastLPAmount, usdcAmount }) => {
+  assertAllDefined({ offerId, fastLPAmount, usdcAmount });
+
+  return {
+    id: offerId,
+    invitationSpec: {
+      source: 'agoricContract',
+      instancePath: ['fastUsdc'],
+      callPipe: [['makeDepositInvitation']],
+    },
+    /** @type {USDCProposalShapes['deposit']} */
+    // @ts-expect-error https://github.com/Agoric/agoric-sdk/issues/10491
+    proposal: {
+      give: {
+        USDC: {
+          brand: brand.USDC,
+          value: usdcAmount,
+        },
+      },
+      want: {
+        PoolShare: {
+          brand: brand.FastLP,
+          value: fastLPAmount,
+        },
+      },
+    },
+  };
+};
+
+/**
+ * @param {Pick<
+ *   import('@agoric/vats/tools/board-utils.js').AgoricNamesRemotes,
+ *   'brand'
+ * >} agoricNames
+ * @param {object} opts
+ * @param {string} opts.offerId
+ * @param {bigint} opts.fastLPAmount
+ * @param {bigint} opts.usdcAmount
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+const makeWithdrawOffer = (
+  { brand },
+  { offerId, fastLPAmount, usdcAmount },
+) => ({
+  id: offerId,
+  invitationSpec: {
+    source: 'agoricContract',
+    instancePath: ['fastUsdc'],
+    callPipe: [['makeWithdrawInvitation']],
+  },
+  proposal: {
+    give: {
+      PoolShare: {
+        // @ts-expect-error https://github.com/Agoric/agoric-sdk/issues/10491
+        brand: brand.FastLP,
+        value: fastLPAmount,
+      },
+    },
+    want: {
+      USDC: {
+        // @ts-expect-error https://github.com/Agoric/agoric-sdk/issues/10491
+        brand: brand.USDC,
+        value: usdcAmount,
+      },
+    },
+  },
+});
+
+/**
+ * @param {Pick<
+ *   import('@agoric/vats/tools/board-utils.js').AgoricNamesRemotes,
+ *   'brand'
+ * >} agoricNames
+ * @param {object} opts
+ * @param {string} opts.offerId
+ * @param {string} opts.operatorId
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+const makeOperatorOffer = (agoricNames, { offerId, operatorId }) => ({
+  id: offerId,
+  invitationSpec: {
+    source: 'agoricContract',
+    instancePath: ['fastUsdc'],
+    callPipe: [['makeOperatorInvitation', [operatorId]]],
+  },
+  proposal: {},
+});
+
+/**
+ * @satisfies {Record<
+ *   string,
+ *   Record<string, import('@agoric/smart-wallet/src/types.js').OfferMaker>
+ * >}
+ */
+export const Offers = {
+  fastUsdc: {
+    Deposit: makeDepositOffer,
+    Withdraw: makeWithdrawOffer,
+    BecomeOperator: makeOperatorOffer,
```

Comment:
I'd like to see a `WithdrawFees` offer maker factored our of `distribute-fees.core.js`. But I suppose it's reasonable to treat core eval stuff differently.

https://github.com/Agoric/agoric-sdk/pull/10735#discussion_r1929480733